server/src/main/java/org/opensearch/search/aggregations/metrics/CardinalityAggregatorFactory.java

```diff
@@ -104,6 +104,9 @@ public static void registerAggregators(ValuesSourceRegistry.Builder builder) {
 
     @Override
     protected Aggregator createUnmapped(SearchContext searchContext, Aggregator parent, Map<String, Object> metadata) throws IOException {
+        if (searchContext.isStreamSearch()) {
```

Comment:
we also need to check flush mode as null or per segment before deciding on `StreamCardinalityAggregator`

Reply:
will fix it in a separate PR